src/preprocess/preprocess.py

```diff
@@ -49,66 +72,117 @@ def remove_all(list_uid, f):
+
+                if rule[action] == remove_rule:
+                    if has_datasets:
+                        generate_delete_datasets_rules(datasets, data_elements, org_units,
+                                                       org_unit_descendants, departament[key][dataset_type], f)
+                    if has_tracker_program:
+                        generate_delete_tracker_rules(tracker_program, data_elements, org_units,
+                                                      org_unit_descendants, departament[key][tracker_type], f)
+                    if has_event_program:
+                        generate_delete_event_rules(event_program, data_elements, org_units,
+                                                    org_unit_descendants, departament[key][program_type], f)
+                    # if not have specific rules apply to all the departament
+                    if not has_datasets and not has_event_program and not has_tracker_program:
+                        remove_all(departament[key], f)
+                elif rule[action] == anonymize_rule:
+                    if has_datasets:
+                        generate_anonymize_datasets_rules(datasets, org_units, data_elements,
+                                anonimize_org_units, anonimize_phone, anonimize_mail,
+                                anonimize_coordinate, departament[key][dataset_type], f)
+                    if has_tracker_program:
+                        generate_anonymize_tracker_rules(tracker_program, tracker_entity_attributes, org_units, data_elements,
+                                anonimize_org_units, anonimize_phone, anonimize_mail,
+                                anonimize_coordinate,
+                                                         departament[key][tracker_type], f)
+                    if has_event_program:
+                        generate_anonymize_event_rules(event_program, org_units, data_elements,
+                                anonimize_org_units, anonimize_phone, anonimize_mail,
+                                anonimize_coordinate,
+                                                       departament[key][program_type], f)
+                    # if not have specific rules apply to all the departament
+                    if not has_datasets and not has_event_program and not has_tracker_program:
+                        for key_type in departament[key].keys():
+                            if key_type == program_type:
+                                generate_anonymize_event_rules(event_program, org_units, data_elements,
+                                                               anonimize_org_units, anonimize_phone, anonimize_mail,
+                                                               anonimize_coordinate,
+                                                               departament[key][program_type], f)
+                            if key_type == dataset_type:
```

Comment:
[minor] When disjoint comparisons, `elif` is more declarative.

Reply:
done